lib/auth/keystore/health_test.go

```diff
@@ -0,0 +1,64 @@
+package keystore
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"testing"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPassiveHealthCheck(t *testing.T) {
```

Comment:
could you add a test case with a non-nil error that should return false for `isHealthCheckFailure`?